src/Lykke.Job.BlockchainTransactionsHistoryDetector/Workflow/Processes/NewTransactionDetectionProcess.cs

```diff
@@ -0,0 +1,247 @@
+                        var transactions = (await client.GetHistoryOfIncomingTransactionsAsync
+                        (
+                            address: wallet.WalletAddress,
+                            afterHash: afterHash,
+                            take: 100,
+                            assetAccuracyProvider: assetId => GetAssetAccuracy(wallet.BlockchainType, assetId)
+                        )).Where(x => x.Hash != afterHash).ToList();
+
+                        foreach (var transaction in transactions)
+                        {
+                            _eventPublisher.PublishEvent(new TransactionDetectedEvent
+                            {
+                                Amount = transaction.Amount,
+                                AssetId = transaction.AssetId,
+                                BlockchainType = wallet.BlockchainType,
+                                FromAddress = transaction.FromAddress,
+                                Hash = transaction.Hash,
+                                Timestamp = transaction.Timestamp,
+                                ToAddress = transaction.ToAddress,
+                                TransactionType = transaction.TransactionType
+                            });
+
+                            afterHash = transaction.Hash;
+                        }
+
+                        taskCompleted = transactions.Count == 0;
+
+                    } while (!taskCompleted && !cancellationToken.IsCancellationRequested);
+
+                    if (wallet.LatestProcessedHash != afterHash)
+                    {
+                        await _observableWalletsRepository.UpdateLatestProcessedHashAsync
+                        (
+                            blockchainType: wallet.BlockchainType,
+                            walletAddress: wallet.WalletAddress,
+                            walletAssetId: wallet.WalletAssetId,
+                            hash: afterHash
+                        );
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                await _log.WriteWarningAsync
+                (
+                    component: nameof(NewTransactionDetectionProcess),
+                    process: nameof(DetectNewTransactionsForWalletAsync),
+                    context: null,
+                    info: null,
+                    ex: e
+                );
+            }
+        }
+
+        private int GetAssetAccuracy(string blockchainType, string assetId)
+        {
+            var asset = _assetsCache.GetOrAdd
+            (
+                $"{blockchainType}-{assetId}",
+                key => _blockchainApiClients[blockchainType].GetAssetAsync(assetId).Result
+            );
+
+            return asset.Accuracy;
+        }
+
+        private async Task GetClientsForSupportedBlockchainsAsync()
+        {
+            if (_blockchainApiClients == null)
+            {
+                var emptyLog = new EmptyLog();
```

Comment:
I think real log could be useful. You could just write to the log that you start probing. In case of some issues we could see error response from the blockchain API in the log, this could help.